src/controllers/metadata.js

```diff
@@ -1,21 +1,22 @@
 'use strict'
 const config = require('../services/config')
 
+const base = config.getIn(['server', 'base_uri'])
 const metadata = {
   currency_code: config.getIn(['currency', 'code']),
   currency_symbol: config.getIn(['currency', 'symbol']),
   condition_sign_public_key: config.getIn(['keys', 'ed25519', 'public']),
   notification_sign_public_key: config.getIn(['keys', 'notification_sign', 'public']),
   urls: {
-    health: '/health',
-    transfer: '/transfers/:id',
-    transfer_fulfillment: '/transfers/:id/fulfillment',
-    transfer_state: '/transfers/:id/state',
-    connectors: '/connectors',
-    accounts: '/accounts',
-    account: '/accounts/:name',
-    subscription: '/subscriptions/:id',
-    subscription_notification: '/subscriptions/:subscription_id/notifications/:notification_id'
+    health: base + '/health',
```

Comment:
The ledger already has to know its public URI, because it uses it in transfer, account, etc. IDs. So base_uri is guaranteed to be the public URI after all proxies etc.

This is currently configured using the `LEDGER_PUBLIC_PATH`, `LEDGER_HOSTNAME`, `LEDGER_PUBLIC_PORT`, `LEDGER_PUBLIC_HTTPS` (from memory) config options. This grew over time, I'd say at some point we should probably just remove all these options and replace them with a `LEDGER_PUBLIC_URI` option.

Note that the previous code (uris starting with /) actually didn't make sense for red/blue.ilpdemo.org because the ledger there is running at a subdirectory: https://red.ilpdemo.org/ledger/

So `/health` would actually mean `https://red.ilpdemo.org/health` (wrong), not `https://red.ilpdemo.org/ledger/health` (correct). This code will always return the complete, correct URI.

Relative URIs are harder to handle for the client because they need to know how to resolve them, and support for relative URIs is only really mature in the HTML/browser context, not in a JSON/API context. (Unless you use JSON-LD, which may be a future improvement we want to look into.)